atmcfg/atmcfg_test.go

```diff
@@ -62,6 +63,130 @@ func TestStartup(t *testing.T) {
+
+		err := ShutdownProcessesByClusterName(config, clusterName, []string{"hostTest:21021"})
+		expected := fmt.Sprintf(`process not found in "%v": %s`, clusterName, "[hostTest:21021]")
+
+		if err == nil {
+			t.Errorf("TestShutdownClusterAndProcesses\n Expected an error but got nil")
+		} else if err.Error() != expected {
+			t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", err.Error(), expected)
+		}
+
+		for i := range config.Processes {
+			if !config.Processes[i].Disabled {
+				t.Errorf("TestShutdownClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[i].Disabled, false)
+			}
+		}
+	})
+}
+
+func TestStartupClusterAndProcesses(t *testing.T) {
+	t.Run("replica set", func(t *testing.T) {
+		config := automationConfigWithOneReplicaSet(clusterName, true)
+		_ = StartupProcessesByClusterName(config, clusterName, []string{"host0:27017"})
+		if config.Processes[0].Disabled {
+			t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, false)
+		}
+	})
+	t.Run("sharded cluster - one process", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+		_ = StartupProcessesByClusterName(config, clusterName, []string{"host2:27018"})
+		if !config.Processes[0].Disabled {
+			t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, true)
+		}
+
+		if config.Processes[1].Disabled {
+			t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[1].Disabled, false)
+		}
+	})
+
+	t.Run("sharded cluster - two processes", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+		_ = StartupProcessesByClusterName(config, clusterName, []string{"host0:27017", "host2:27018"})
+		if config.Processes[0].Disabled {
+			t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[0].Disabled, false)
+		}
+
+		if config.Processes[1].Disabled {
+			t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[1].Disabled, false)
+		}
+	})
+	t.Run("startup entire sharded cluster", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+
+		_ = StartupProcessesByClusterName(config, clusterName, nil)
+		for i := range config.Processes {
+			if config.Processes[i].Disabled {
+				t.Errorf("TestStartupClusterAndProcesses\n got=%#v\nwant=%#v\n", config.Processes[i].Disabled, false)
+			}
+		}
+	})
+	t.Run("provide a process that does not exist", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+
+		err := StartupProcessesByClusterName(config, clusterName, []string{"hostTest:21021"})
+		expected := fmt.Sprintf(`process not found in "%v": %s`, clusterName, "[hostTest:21021]")
+
+		if err == nil {
+			t.Errorf("TestStartupClusterAndProcesses\n Expected an error but got nil")
+		} else if err.Error() != expected {
```

Comment:
[nit] this is bad error handling for modern go, please check https://blog.golang.org/go1.13-errors